src/components/Filter/FilterValueSelector.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { DropdownButton } from '../Button';
+import { MenuItem } from '../MenuItem';
+import { bindMethods } from '../../common/helpers';
+import cx from 'classnames';
+
+class FilterValueSelector extends React.Component {
+  constructor() {
+    super();
+    bindMethods(this, ['selectFilterValue']);
+  }
+
+  selectFilterValue(filterValue) {
+    const { onFilterValueSelected } = this.props;
+
+    onFilterValueSelected && onFilterValueSelected(filterValue);
+  }
+
+  renderPlaceHolder() {
+    const { placeholder } = this.props;
+
+    if (!placeholder) {
+      return null;
+    }
+
+    return (
+      <MenuItem
+        title={placeholder}
+        key="Placeholder"
+        onSelect={() => this.selectFilterValue()}
+      >
+        {placeholder}
+      </MenuItem>
+    );
+  }
+
+  renderSelectMenuItems() {
+    const { filterValues, currentValue } = this.props;
+
+    return filterValues.map((item, index) => {
+      let classes = {
+        selected: item === currentValue
+      };
+      return (
+        <MenuItem
+          className={classes}
+          key={item.id || index}
+          onSelect={() => this.selectFilterValue(item)}
+        >
+          {item.title || item}
+        </MenuItem>
+      );
+    });
+  }
+
+  render() {
+    const {
+      className,
+      id,
+      filterValues,
+      currentValue,
+      placeholder
+    } = this.props;
+    let classes = cx('filter-pf-select', className);
+
+    if (placeholder || (filterValues && filterValues.length > 1)) {
+      let title;
+      if (currentValue) {
+        title = currentValue.title || currentValue;
+      } else {
+        title = placeholder || filterValues[0].title || filterValues[0];
+      }
+
+      let menuId = 'filterCategoryMenu';
+      menuId += id ? '_' + id : '';
```

Comment:
Maybe a template literal with backticks... think there are a couple of these
```
menuId += id ? `_${id}` : ...
```

Reply:
Fixed. Thanks.